doc/port-si/port-si-per-speed.md

```diff
@@ -0,0 +1,267 @@
+
+    Before transitioning to independent mode, SERDES SI parameters were never applied because the media_settings.json file was not found on the platform. In the new mode, media_settings.json will be present on every platform, but it will only be utilized when it contains an entry for a plugged-in CMIS module (QSFP-DD, OSFP). If it does not have a suitable entry for the connected module, nothing will be written to APP_DB to maintain backward compatibility with the legacy flow.
+
+    In order for the FW to configure the ASIC in the case of non-CMIS modules, instead of SONiC, it is crucial to ensure that the           media_settings.json file satisfies the following two requirements:
+    - The media_settings.json file should only contain data related to CMIS modules. Signal Integrity (SI) parameters for non-CMIS modules should not be present in this JSON file.
+
+    - The media_settings.json file must not include a 'default' section. Normally, the lookup process in the media_settings.json file allows for a default section. SI parameters, defined in this section, will be written to the APP_DB if there is no matching entry found for any other module in the JSON file. However, since we do not intend SONiC configuration to support non-CMIS modules, it is necessary for the media_settings.json file to exclude the default section.
+
+    </br>Ports OrchAgent listens to changes in APP_DB, so when the APP_DB is updated with SI parameters, PortsOrchagent is triggered. Based on the data found in APP_DB, PortsOA creates a vector that contains the SI values for a certain port and passes it as a whole to the SAI. Eventually, they will write it to the SLTP register in PHY upon receiving ADMIN_UP.  
+  </br></br></br>
+
+
+## Port SI configuration (flows)
+
+![Port SI Flows](<Port SI Flows.drawio.svg>)  
+*** Risk: Currently the loading of media_settings.json is skipped in case fast-reboot is enabled. There's an open issue that is under discussion with Microsoft to revert this commit: https://github.com/sonic-net/sonic-platform-daemons/pull/221. 
+
+As described above, passing the port SI parameters is carried out in 3 scenarios:
+
+- Initialization phase: When the switch is going up, the **_Notify-Media-setting-Process_** is carried out for each of the current interfaces. 
+- Module Plug in event
+- Port speed change event: When there is a speed change (port-breakout for example), it is going to affect the speed per lane </br></br></br></br>
+
+
+# Changes to support the new mode
+
+1. The XCVRD::Notifty_media_settings() function should be modified to support the new SI parameters and the format of media_settings_json:
+
+   - The method get_media_settings_key() should be extended:
+
+      We need to extend the key used for lookup in the media_settings.json file to include the lane speed as well. Currently, the key is composed of three components: media type, specification compliance, and length.
```

Comment:
Can you please mention the logic used to compute key based on specification compliance? How will it work when the extended compliance has more values
        Specification compliance:
                10/40G Ethernet Compliance Code: Unknown
                Extended Specification Compliance: 25GBASE-CR CA-25G-N or 50GBASE-CR2 with no FEC

Here is an example for QSFP DD port
Ethernet200: SFP EEPROM detected
        Active Firmware: 0.0
        Active application selected code assigned to host lane 1: N/A
        Active application selected code assigned to host lane 2: N/A
        Active application selected code assigned to host lane 3: N/A
        Active application selected code assigned to host lane 4: N/A
        Active application selected code assigned to host lane 5: N/A
        Active application selected code assigned to host lane 6: N/A
        Active application selected code assigned to host lane 7: N/A
        Active application selected code assigned to host lane 8: N/A
        Application Advertisement: 400G CR8 - Host Assign (0x1) - Copper cable - Media Assign (Unknown)
                                   200GBASE-CR4 (Clause 136) - Host Assign (0x11) - Copper cable - Media Assign (Unknown)
                                   100GBASE-CR2 (Clause 136) - Host Assign (0x55) - Copper cable - Media Assign (Unknown)
                                   100GBASE-CR4 (Clause 92) - Host Assign (0x11) - Copper cable - Media Assign (Unknown)
                                   50GBASE-CR (Clause 126) - Host Assign (0xff) - Copper cable - Media Assign (Unknown)
                                   40GBASE-CR4 (Clause 85) - Host Assign (0x11) - Copper cable - Media Assign (Unknown)
                                   25GBASE-CR CA-N (Clause 110) - Host Assign (0xff) - Copper cable - Media Assign (Unknown)
                                   1000BASE -CX(Clause 39) - Host Assign (0xff) - Copper cable - Media Assign (Unknown)
        CMIS Rev: 3.0
        Connector: No separable connector
        Encoding: N/A
        Extended Identifier: Power Class 1 (0.25W Max)
        Extended RateSelect Compliance: N/A
        Host Lane Count: 8
        Identifier: QSFP-DD Double Density 8X Pluggable Transceiver
        Inactive Firmware: N/A
        Length Cable Assembly(m): 0.5
        Media Interface Technology: Copper cable unequalized
        Media Lane Count: 0
        Module Hardware Rev: 0.0
        Nominal Bit Rate(100Mbs): 0
        Specification compliance: passive_copper_media_interface
        Supported Max Laser Frequency: N/A
        Supported Max TX Power: N/A
        Supported Min Laser Frequency: N/A
        Supported Min TX Power: N/A
        Vendor Date Code(YYYY-MM-DD Lot): 2020-04-14
        Vendor Name: Mellanox
        Vendor OUI: 00-02-c9
        Vendor PN: MCP1660-W00AE30
        Vendor Rev: A2
        Vendor SN: MT2016VS01995